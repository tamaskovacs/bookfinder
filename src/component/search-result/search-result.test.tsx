import React from 'react';
import { render } from '@testing-library/react';
import SearchResult from './search-result';

describe('<SearchResult /> ', () => {
    it('should match the snapshot if has result', () => {
        const rowData = [{
            title: 'test title',
            cover_edition_key: '12345',
            author_name: 'test author',

        },{
            title: 'test title2 ',
            cover_edition_key: '12345 22',
            author_name: 'test author 2',

        }];

        const elementFilled = render(<SearchResult
            rowData={rowData}
            selectionHandler={()=>jest.fn()}
        />);
        expect(elementFilled).toMatchSnapshot();
    });

    it('should match the snapshot if has no result', () => {
        const rowData:Array<any> = [];

        const elementEmpty = render(<SearchResult
            rowData={rowData}
            selectionHandler={()=>jest.fn()}
        />);
        expect(elementEmpty).toMatchSnapshot();
    });
});
