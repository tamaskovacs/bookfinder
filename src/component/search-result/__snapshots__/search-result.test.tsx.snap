// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult />  should match the snapshot if has no result 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="table"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="table"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<SearchResult />  should match the snapshot if has result 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="table"
      >
        <div
          class="row"
        >
          <div
            class="cover-cell"
          >
            <img
              alt="test title"
              class="cover"
              src="http://covers.openlibrary.org/b/olid/12345-M.jpg"
            />
          </div>
          <div
            class="cell"
          >
            <div
              class="author-cell"
            >
              test author
            </div>
            <div
              class="title-cell"
            >
              test title
            </div>
          </div>
          <div
            class="button-cell"
          >
            <button>
              Details
            </button>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="cover-cell"
          >
            <img
              alt="test title2 "
              class="cover"
              src="http://covers.openlibrary.org/b/olid/12345 22-M.jpg"
            />
          </div>
          <div
            class="cell"
          >
            <div
              class="author-cell"
            >
              test author 2
            </div>
            <div
              class="title-cell"
            >
              test title2 
            </div>
          </div>
          <div
            class="button-cell"
          >
            <button>
              Details
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="table"
    >
      <div
        class="row"
      >
        <div
          class="cover-cell"
        >
          <img
            alt="test title"
            class="cover"
            src="http://covers.openlibrary.org/b/olid/12345-M.jpg"
          />
        </div>
        <div
          class="cell"
        >
          <div
            class="author-cell"
          >
            test author
          </div>
          <div
            class="title-cell"
          >
            test title
          </div>
        </div>
        <div
          class="button-cell"
        >
          <button>
            Details
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="cover-cell"
        >
          <img
            alt="test title2 "
            class="cover"
            src="http://covers.openlibrary.org/b/olid/12345 22-M.jpg"
          />
        </div>
        <div
          class="cell"
        >
          <div
            class="author-cell"
          >
            test author 2
          </div>
          <div
            class="title-cell"
          >
            test title2 
          </div>
        </div>
        <div
          class="button-cell"
        >
          <button>
            Details
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
